File: python_implemented/binarySearch.py
#Binary Search using recursion

def binarySearch(arr, x, l, r):
	if l == r:
		if x == arr[l]:
			return l
		else:
			return -1
	else:
		mid = (l + r) / 2
		if x <= arr[mid]:
			return binarySearch(arr, x, l, mid)
		else:
			return binarySearch(arr, x, mid+1, r)

def main():
	arr = []
	n = input("Enter the length: ")
	var = raw_input("Enter the elements: ")
	arr = map(int, var.split())
	x = input("Enter the element to be searched: ")
	pos = binarySearch(arr, x, 0, n-1)

	if pos > 0:
		print "Position of %d is %d .\n" % (x, pos + 1)
	else:
		print "Element not found.\n"

main()

File: C-C++/binarySearch.cpp
#include<iostream>

int binary_search(int k,int a[],int n)
{   int end=n-1,start=0;
while(start<=end)
{	 int m=(end+start)/2;
        if(k==a[m])
           return 1;

        else if(k<a[m])
            end=m-1;
        else if(k>a[m])
            start=m+1;
        else
            return 0;
}}
int main()
{   int a[20];
    int i,n,k,f;
    printf("enter the size of array\n");

    scanf("%d",&n);
    for(i=0;i<n;++i)
    scanf("%d",&a[i]);
    printf("enter no. to search");
    scanf("%d",&k);

    f=binary_search(k,a,n);
    if(f==1)
    printf("number is found");
    else
    printf("not found");
    return 0;
}
